File: app/src/main/java/me/wcy/music/utils/SystemUtils.java
package me.wcy.music.utils;

import android.app.ActivityManager;
import android.app.Notification;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.text.format.DateUtils;

import java.util.List;
import java.util.Locale;

import me.wcy.music.R;
import me.wcy.music.activity.SplashActivity;
import me.wcy.music.constants.Extras;
import me.wcy.music.model.Music;

/**
 * Created by hzwangchenyan on 2016/3/22.
 */
public class SystemUtils {

    /**
     * 判断是否有Activity在运行
     */
    public static boolean isStackResumed(Context context) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        List<ActivityManager.RunningTaskInfo> runningTaskInfos = manager.getRunningTasks(1);
        ActivityManager.RunningTaskInfo runningTaskInfo = runningTaskInfos.get(0);
        return runningTaskInfo.numActivities > 1;
    }

    /**
     * 判断Service是否在运行
     */
    public static boolean isServiceRunning(Context context, Class<?> serviceClass) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (serviceClass.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }

    public static Notification createNotification(Context context, Music music) {
        String title = music.getTitle();
        String subtitle = FileUtils.getArtistAndAlbum(music.getArtist(), music.getAlbum());
        Bitmap cover = CoverLoader.getInstance().loadThumbnail(music);
        Intent intent = new Intent(context, SplashActivity.class);
        intent.putExtra(Extras.FROM_NOTIFICATION, true);
        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
        Notification.Builder builder = new Notification.Builder(context)
                .setContentIntent(pendingIntent)
                .setContentTitle(title)
                .setContentText(subtitle)
                .setSmallIcon(R.drawable.ic_notification)
                .setLargeIcon(cover);
        return builder.getNotification();
    }

    public static String formatTime(String pattern, long milli) {
        int m = (int) (milli / DateUtils.MINUTE_IN_MILLIS);
        int s = (int) ((milli / DateUtils.SECOND_IN_MILLIS) % 60);
        String mm = String.format(Locale.getDefault(), "%02d", m);
        String ss = String.format(Locale.getDefault(), "%02d", s);
        return pattern.replace("mm", mm).replace("ss", ss);
    }
}

File: app/src/main/java/me/wcy/music/utils/SystemUtils.kt
package me.wcy.music.utils

import android.app.ActivityManager
import android.app.Service
import android.content.Context
import android.text.TextUtils
import android.text.format.DateUtils
import java.util.Locale

/**
 * Created by hzwangchenyan on 2016/3/22.
 */
object SystemUtils {
    /**
     * 判断是否有Activity在运行
     */
    fun isStackResumed(context: Context): Boolean {
        val manager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val runningTaskInfos = manager.getRunningTasks(1)
        val runningTaskInfo = runningTaskInfos[0]
        return runningTaskInfo.numActivities > 1
    }

    /**
     * 判断Service是否在运行
     */
    fun isServiceRunning(context: Context, serviceClass: Class<out Service?>): Boolean {
        val manager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        for (service in manager.getRunningServices(Int.MAX_VALUE)) {
            if (serviceClass.name == service.service.className) {
                return true
            }
        }
        return false
    }

    val isFlyme: Boolean
        get() {
            val flymeFlag = getSystemProperty("ro.build.display.id")
            return !TextUtils.isEmpty(flymeFlag) && flymeFlag!!.lowercase(Locale.getDefault())
                .contains("flyme")
        }

    private fun getSystemProperty(key: String): String? {
        try {
            val classType = Class.forName("android.os.SystemProperties")
            val getMethod = classType.getDeclaredMethod("get", String::class.java)
            return getMethod.invoke(classType, key) as String
        } catch (th: Throwable) {
            th.printStackTrace()
        }
        return null
    }

    fun formatTime(pattern: String, milli: Long): String {
        val m = (milli / DateUtils.MINUTE_IN_MILLIS).toInt()
        val s = (milli / DateUtils.SECOND_IN_MILLIS % 60).toInt()
        val mm = String.format(Locale.getDefault(), "%02d", m)
        val ss = String.format(Locale.getDefault(), "%02d", s)
        return pattern.replace("mm", mm).replace("ss", ss)
    }
}